nni/algorithms/compression/pytorch/quantization/quantizers.py

```diff
@@ -545,4 +545,81 @@ def export_model(self, model_path, calibration_path=None, onnx_path=None, input_
 
         self.export_model_save(self.bound_model, model_path, calibration_config, calibration_path, onnx_path, input_shape, device)
 
-        return calibration_config
\ No newline at end of file
+        return calibration_config
+
+
+class LsqQuantizer(Quantizer):
+    """Quantizer defined in:
+       Learned Step Size Quantization (ICLR 2020)
+       https://arxiv.org/pdf/1902.08153.pdf
+       """
+
+    def __init__(self, model, config_list, optimizer=None):
+        super().__init__(model, config_list, optimizer)
+        self.quant_grad = QuantForward()
```

Comment:
If we keep the original forward and backward structure, the Lsq can forward as usual and backward by **STE**. In this way, will it be anything wrong? May be have something to do with the update of scale and zeropoint.

Reply:
There will not be anything wrong if the gradients are handled carefully. However, there exists one major limitation for the origin framework, that is, we must customize all gradients for all learnable parameters. If the gradient-based algorithms become complex, it will be troubling and  error-prone to do the customization. In this situation, I think using the auto-grad system to determine the gradient is more convenient for users.